core/src/cost_model.rs

```diff
@@ -450,4 +473,33 @@ mod tests {
             th.join().unwrap();
         }
     }
+
+    #[test]
+    fn test_cost_model_init_cost_table() {
+        // build cost table
+        let instruction_ids = vec![
+            Pubkey::new_unique(),
+            Pubkey::new_unique(),
+            Pubkey::new_unique(),
+        ];
+        let instruction_costs = vec![10, 20, 30];
+        let cost_table: Vec<(Pubkey, u64)> = instruction_ids
+            .iter()
+            .enumerate()
+            .map(|(index, id)| (*id, instruction_costs[index]))
+            .collect();
+
+        // init cost model
+        let mut cost_model = CostModel::default();
+        cost_model.initialize_cost_table(&cost_table);
+        let cost_model = Arc::new(RwLock::new(cost_model));
+
+        // verify
+        instruction_ids.iter().enumerate().for_each(|(index, id)| {
+            assert_eq!(
+                instruction_costs[index],
+                cost_model.read().unwrap().find_instruction_cost(id)
+            );
+        });
+    }
```

Comment:
Nitpicks
- I think we could simplify the cost table construction, which would in turn simplify the verify step.
- I think we could use `cost_model` as-is without wrapping it into an `Arc<RwLock<CostModel>>`; it looks like the other tests do not wrap the `CostModel` object up either

```suggestion
    fn test_cost_model_init_cost_table() {
        // build cost table
        let cost_table = vec![
            (Pubkey::new_unique(), 10),
            (Pubkey::new_unique(), 20),
            (Pubkey::new_unique(), 30),
        ];
        
        // init cost model
        let mut cost_model = CostModel::default();
        cost_model.initialize_cost_table(&cost_table);

        // verify
        for (id, cost) in cost_table.iter() {
            assert_eq!(*cost, cost_model.find_instruction_cost(id));
        }
    }
```

Reply:
thanks @steviez ... like your suggestions! For #2 regarding wrapping cost_model object, my thought is to mimi actual use case in test, so I wrapped it as it is in client code. It does seem redundant by itself. But #1 makes the test so much cleaner. I'll update the test.